docs/DeveloperGuide.md

```diff
@@ -1,38 +1,252 @@
 # Developer Guide
 
+## Table of Contents
+
+- [Developer Guide](#developer-guide)
+    - [Table of Contents](#table-of-contents)
+    - [Acknowledgements](#acknowledgements)
+    - [Design \& Implementation](#design--implementation)
+        - [UI and I/O](#ui-and-io)
+        - [Commands](#commands)
+        - [Storage](#storage)
+        - [Item and ItemMap](#item-and-itemmap)
+        - [DateTime](#datetime)
+        - [Exceptions and Logging](#exceptions-and-logging)
+    - [Product scope](#product-scope)
+        - [Target user profile](#target-user-profile)
+        - [Value proposition](#value-proposition)
+    - [User Stories](#user-stories)
+    - [Non-Functional Requirements](#non-functional-requirements)
+    - [Glossary](#glossary)
+    - [Instructions for Testing](#instructions-for-testing)
+        - [Manual Testing](#manual-testing)
+        - [JUnit Testing](#junit-testing)
+        - [Text UI Testing](#text-ui-testing)
+    - [Future Enhancements](#future-enhancements)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+<!-- @@author yijiano -->
+
+PILL uses the following tools for development:
+1. [JUnit 5](https://junit.org/junit5/) - Used for testing.
+2. [Gradle](https://gradle.org/) - Used for build automation.
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The project is designed using the Model-View-Controller (MVC) architecture, with the following components:
+
+1. **Model**: Contains the core logic of the application, including the Item, ItemMap, and Storage classes.
+2. **View**: The user interface, which is implemented as a command-line interface (CLI).
+3. **Controller**: The command classes, which interpret user input and call the appropriate methods from the Model and
+   Storage classes.
+4. **Storage**: Handles the reading and writing of data to a CSV file.
+5. **DateTime**: A utility class to handle date and time operations.
+6. **Exceptions and Logging**: Custom exceptions and a logging utility class to manage application-wide logging.
+7. **PillLogger**: A centralized utility class to handle logging across the entire application.
+8. **PillException**: A custom exception class to handle exceptions specific to the PILL application.
+9. **Item**: A class to represent an item in the inventory.
+10. **ItemMap**: A class to store items in a map-like structure.
+11. **Command**: A class to represent a command that can be executed by the application.
+12. **Parser**: A class to parse user input and return the corresponding command.
+13. **Pill**: The main class that initializes the application and starts the CLI.
+
+### High-Level Overview
+
+The high-level overview of the project structure is as follows:
+
+<img src = "diagrams/High-Level-Overview.png" alt="High Level Overview of PILL"/>
```

Comment:
You may try to consider the direction of arrows. Since it's is MVC model, the arrow from View to Controller might be a two way arrow (e.g. View send request to Controller and Controller response with data?), the same for Controller to Model components